doc/api/stream.md

```diff
@@ -774,7 +774,7 @@ end
 *Note*: In general, the `readable.pipe()` and `'data'` event mechanisms are
 easier to understand than the `'readable'` event.
 However, handling `'readable'` might result in increased throughput.
-
+While new data is pushed but not directly from executing `readable.read()`, `'readable'` event will not be triggered.
```

Comment:
The process is slightly more convoluted: after `.read()` returns `null`, the application is supposed to listen for the `'readable'` event to know when to call `.read()`  again. It will be emitted the first time data is pushed.

Reply:
@mcollina Yeah you are right!  I just find that ```.read()``` inside have many methods and attributes including ```_read()``` and ``` needReadable```. If we exectue ```_read()``` directly, we will not execute other code including reset ``` needReadable```'s value. and then new data is available but ``` needReadable``` maybe not change(``` needReadable```decided whether emit  `'readable'` event happened), at last  `'readable'` event will not be triggered